c7n/resources/ebs.py

```diff
@@ -50,8 +50,44 @@ class SnapshotAge(AgeFilter):
         days={'type': 'number'},
         op={'type': 'string', 'enum': OPERATORS.keys()})
     date_attribute = 'StartTime'
-
-
+    
+
+def _filter_ami_snapshots(self, snapshots):
+    c = local_session(self.manager.session_factory).client('ec2')
+    #try using cache first to get a listing of all AMI snapshots and compares resources to the list
+    #This will populate the cache.
+    ami_manager = AMI(self.manager.ctx, {})
+    amis = ami_manager.resources()
+    ami_snaps = []
+    for i in amis:
+        for dev in i.get('BlockDeviceMappings'):
+            if 'Ebs' in dev and 'SnapshotId' in dev['Ebs']:
+                ami_snaps.append(dev['Ebs']['SnapshotId'])            
+    matches = []
+    if not self.data.get('value', True):
```

Comment:
this line should be first, or done by the caller, if its the caller then you can drop self parameter and turn this into an actual function, instead of a pseudo method.